### PR TITLE
Implement custom ruler color for Plot

### PR DESCRIPTION
**Pull request by gweisert**
Implement custom ruler color for Plot which is exposed through the new `custom_ruler_color` method. The entire thing is only a few lines.

### CODE REVIEW

**Comment by bircni** [comment]
This should work, the current implementation doesn't support setting the color BUT it works with light and dark mode - your implementation doesn't work with light and dark mode as it is always the same color
https://github.com/emilk/egui_plot/blob/0fe0ea6cbf0fe01f2b27f9247516fc63dbb25b0f/egui_plot/src/items/mod.rs#L1939

**Comment by gweisert** [comment]
@bircni I'm not quite sure what you are trying to say.
If no custom ruler color is specified, the function you mentioned is still used to determine the color:
```rust
  // before
  let line_color = rulers_color(ui);
  // now
  let line_color = self.ruler_color.unwrap_or_else(|| rulers_color(ui));
```

**Comment by bircni** [comment]
Sure I wanted to say that it might be better to set two colors, one for dark and one for light.
Sorry that was kinda weird how I wrote that

**Comment by gweisert** [comment]
That seems really unnecessaty complication, imo.
If that is your use case, you could easily choose the color before:
```rust
let my_color = if ui.visuals().dark_mode { Color32::RED } else { Color32::GREEN };
egui_plot::Plot::new("my_plot").custom_ruler_color(my_color).show( [...] )
```

File: egui_plot/src/lib.rs [hunk 1]
```diff
@@ -1517,7 +1531,7 @@ impl<'a> PreparedPlot<'a> {
         };
 
         // Draw cursors
-        let line_color = rulers_color(ui);
+        let line_color = self.ruler_color.unwrap_or_else(|| rulers_color(ui));
```
Comment (emilk) [inline egui_plot/src/lib.rs:1534]: What you call "ruler" is called "cursor" elsewhere

Comment (gweisert) [reply egui_plot/src/lib.rs:1534]: the original function is called `rulers_color`, that's why I chose that name 😅
I'll change it 👍

**Comment by bircni** [comment]
@emilk this seems to be ready to merge

File: egui_plot/src/lib.rs [hunk 2]
```diff
@@ -727,6 +729,15 @@ impl<'a> Plot<'a> {
         self
     }
 
+    /// Set custom cursor color.
+    ///
+    /// You may set the color to ``Color32::TRANSPARENT`` to hide the cursors.
```
Comment (emilk) [inline egui_plot/src/lib.rs:734]:
```suggestion
    /// You may set the color to [`Color32::TRANSPARENT`] to hide the cursors.
```

File: egui_plot/src/lib.rs [hunk 3]
```diff
@@ -727,6 +729,15 @@ impl<'a> Plot<'a> {
...
+    #[inline]
+    pub fn custom_cursor_color(mut self, color: Color32) -> Self {
```
Comment (emilk) [inline egui_plot/src/lib.rs:736]:
```suggestion
    pub fn cursor_color(mut self, color: Color32) -> Self {
```

**Review by emilk — APPROVED** [review]

**Pull request merged**